Math/NumberTheory/Zeta/Hurwitz.hs

```diff
@@ -0,0 +1,86 @@
+-- |
+-- Module:      Math.NumberTheory.Zeta.Hurwitz
+-- Copyright:   (c) 2018 Alexandre Rodrigues Baldé
+-- Licence:     MIT
+-- Maintainer:  Alexandre Rodrigues Baldé <alexandrer_b@outlook.com>
+-- Stability:   Provisional
+-- Portability: Non-portable (GHC extensions)
+--
+-- Hurwitz zeta function.
+
+{-# LANGUAGE ScopedTypeVariables #-}
+
+module Math.NumberTheory.Zeta.Hurwitz
+  ( zeta
+  ) where
+
+import Data.List                      (zipWith4)
+
+import Math.NumberTheory.Recurrencies (bernoulli, factorial)
+import Math.NumberTheory.Zeta.Utils   (skipOdds, suminf)
+
+-- | Value of Hurwitz zeta function evaluated at @ζ(s, a)@ with
+-- @forall t1 t2 . (Floating t1, Ord t1, Integral t2) => s ∈ t2, a ∈ t1@.
+-- The algorithm used was based on the Euler-Maclaurin formula and was derived
+-- from the <https://dlmf.nist.gov/25.11#iii Digital Library of Mathematical Functions>
+-- by the <https://www.nist.gov/ National Institute of Standards and Technology (NIST)>,
+-- formula 25.11.5.
+zeta :: forall a b . (Floating a, Ord a, Integral b) => a -> b -> a -> a
```

Comment:
Let us name it `zetaHurwitz` or something else, different from `zeta`. Just to avoid confusion with Riemann zeta.